### PR TITLE
password reset to reflect new password

### PR DESCRIPTION
**Pull request by adamjezek98**

### CODE REVIEW

File: src/services/backend/users-me.js [hunk 1]
```diff
@@ -83,7 +83,7 @@ export const authenticateFresh = (username, password) => {
       username,
       password,
     },
-    successCodes: [200],
+    successCodes: [400],
```
Comment (czervenka) [inline src/services/backend/users-me.js:86]: If this function succeeds on "400 Bad request" response, why you just don't remove it?

File: src/reducers/users-me.js [hunk 2]
```diff
@@ -88,12 +88,7 @@ export default (
       persistUserProfile(userData);
       return Object.assign({}, state, userData);
     case "USER_CHANGE_PASSWORD_SUCCEEDED":
-      userData = getUserDataFromApiResponse(
-        action.payload.data,
-        state.activeOrganization
-      );
-      persistUserProfile(userData);
-      return Object.assign({}, state, userData);
+     return state;
```
Comment (czervenka) [inline src/reducers/users-me.js:91]: Return state is the default action. You can IMO safely remove this lines.

File: src/services/backend/users-me.js [hunk 3]
```diff
@@ -122,14 +122,14 @@ export const changePassword = (
   new_password_confirmation
```
Comment (czervenka) [inline src/services/backend/users-me.js:122]: new_password_confirmation is not used here any more, right?

File: src/actions/users-me.js [hunk 4]
```diff
@@ -140,12 +140,8 @@ export const changePassword = createHttpAction(
     password,
     new_password,
     new_password_confirmation,
-    { dispatch, getState }
   ) => {
-    const { me } = getState();
-    return dispatch(authenticateFresh(me.username, password)).then((r) =>
-      backend.changePassword(password, new_password, new_password_confirmation)
-    );
+     return backend.changePassword(password, new_password, new_password_confirmation)
```
Comment (czervenka) [inline src/actions/users-me.js:144]: `new_password_confirmation` is not validated on backend now, is it validated somewhere on FE?

**Review by czervenka — CHANGES_REQUESTED** [review]

File: src/actions/users-me.js (hunk 4 above)
Comment (adamjezek98) [reply src/actions/users-me.js:138]: yes, the form validates it and shows error message if there is a mismatch. no request to backend is sent

File: src/services/backend/users-me.js (hunk 1 above)
Comment (adamjezek98) [reply src/services/backend/users-me.js]: I just changed that somewhere in the process to work around that and then forgot about it.

**Pull request merged**